airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/jsonschema/AirbyteJsonSchemaType.kt

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.integrations.destination.s3.jsonschema
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.databind.node.ObjectNode
+
+enum class AirbyteJsonSchemaType {
+    BOOLEAN,
+    INTEGER,
+    NUMBER,
+    STRING,
+    BINARY_DATA,
+    DATE,
+    TIMESTAMP_WITH_TIMEZONE,
+    TIMESTAMP_WITHOUT_TIMEZONE,
+    TIME_WITH_TIMEZONE,
+    TIME_WITHOUT_TIMEZONE,
+    ARRAY_WITHOUT_ITEMS,
+    ARRAY_WITH_ITEM,
+    ARRAY_WITH_ITEMS,
+    OBJECT_WITHOUT_PROPERTIES,
+    OBJECT_WITH_PROPERTIES,
+    UNION,
+    COMBINED;
+
+    fun matchesValue(tree: JsonNode): Boolean {
+        return when (this) {
+            BOOLEAN -> tree.isBoolean
+            INTEGER -> tree.isIntegralNumber || tree.isInt || tree.isBigInteger
+            NUMBER ->
+                tree.isDouble ||
+                    tree.isFloat ||
+                    tree.isIntegralNumber ||
+                    tree.isInt ||
+                    tree.isBigInteger
+            STRING -> tree.isTextual
+            BINARY_DATA -> tree.isBinary || tree.isTextual
+            DATE,
+            TIMESTAMP_WITH_TIMEZONE,
+            TIMESTAMP_WITHOUT_TIMEZONE,
+            TIME_WITH_TIMEZONE,
+            TIME_WITHOUT_TIMEZONE -> tree.isTextual
+            ARRAY_WITHOUT_ITEMS,
+            ARRAY_WITH_ITEM,
+            ARRAY_WITH_ITEMS -> tree.isArray
+            OBJECT_WITHOUT_PROPERTIES -> tree.isObject
+            OBJECT_WITH_PROPERTIES -> tree.isObject
+            UNION,
+            COMBINED -> throw IllegalStateException("Union type cannot be matched")
+        }
+    }
+
+    companion object {
+        fun fromJsonSchema(schema: ObjectNode): AirbyteJsonSchemaType {
+            if (schema.has("oneOf")) {
+                return UNION
+            }
+
+            val ref = schema["\$ref"]?.asText()
+            if (ref != null) {
+                return when (ref) {
+                    "WellKnownTypes.json#/definitions/Boolean" -> BOOLEAN
```

Comment:
It's in a lot of the existing test data I can drop it across the board when I clean that up for the CDK work & merge this with the existing airbyte schema type. I skipped it before because I couldn't verify there were 0 schemas using it (querying the configdb takes too long :).)

Reply:
I can guarantee that nobody is using it in production :fire: (there's no support for it in platform/destinations, I'm pretty sure it only exists in test code / dead runtime code)